### PR TITLE
Lower overhead of Windows CI tests

### PR DESCRIPTION
**Pull request by mlxd**
### Before submitting

Please complete the following checklist when submitting a PR:

- [x] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [x] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:** The Windows test suite often takes a long time to complete, or hangs on coverage generation during tests. Improving the Windows CI test runtime will help unblock other PRs in the ecosystem, where Windows CI runners are often timing out.

**Description of the Change:** This PR removes system-level debug files from the OpenCPPCoverage PDB paths, caches the vcpkg generated builds, and uses a RelWithDebInfo build for the coverage tests under Windows, following support in OpeCPPCoverage.

**Benefits:** Unblocks the Windows CI tests of incoming PRs.

**Possible Drawbacks:** Time-out issue may still occur, though frequency of occurrence dropped during testing of this PR.

**Related GitHub Issues:**

### CODE REVIEW

**Comment by mlxd** [comment]
Note that the CUDA 12 PR is required to pass the final tests, so this will first be merged onto that.

**Review by AmintorDusko — APPROVED** [review]
Beautiful! Thank you for that!

File: .github/workflows/tests_windows.yml [hunk 1]
```diff
@@ -163,9 +93,111 @@ jobs:
...
+    timeout-minutes: 30
+    name: vcpkg dependencies
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Cache installation directories
+        id: vcpkg-cache
+        uses: actions/cache@v4
+        with:
+          path: D:\a\install_dir\vcpkg
```
Comment (maliasadi) [inline .github/workflows/tests_windows.yml:110]: nice! I guessed there is more work required for caching vcpkg when suggested to use that action :smile: the cached folder is quite large (830 MB)!

**Review by maliasadi — APPROVED** [review]
:raised_hands:

**Pull request merged**